test/helpers/externaldns.go

```diff
@@ -20,19 +28,30 @@ func RecordSets(d, hostedZone string) (bool, error) {
 		HostedZoneId: aws.String(hostedZone),
 	}
 
-	sets, err := svc.ListResourceRecordSets(params)
-	if err != nil {
-		return false, err
-	}
+	wait := startWaitTime
 
-	for _, v := range sets.ResourceRecordSets {
-		record := strings.TrimRight(*v.Name, ".")
-		if record == d {
-			return true, nil
+	for i := 0; i < maxRetries; i++ {
+		sets, err := svc.ListResourceRecordSets(params)
+		if err != nil {
+			awsErr, ok := err.(awserr.Error)
+			if ok && awsErr.Code() == "Throttling" {
+				time.Sleep(wait)
+				wait *= 2 // exponential backoff
```

Comment:
is it waiting for 2 sec?

Reply:
this is the factoring value, it'll increase the wait on each failed pass:
1,2,4,8,16.

trying out AWS advice as per:
https://repost.aws/knowledge-center/route-53-avoid-throttling-errors